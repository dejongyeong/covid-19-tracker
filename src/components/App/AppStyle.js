import styled from "styled-components";

/* eslint-disable */
export const Wrapper = styled.div`
  color: #ffffff;
  width: auto;
  height: 100%;
  position: relative;
  overflow: hidden;

  @media only screen and ${(props) => props.theme.breakpoints.lg} and (orientation: portrait) {
    overflow: none
  }
`;
