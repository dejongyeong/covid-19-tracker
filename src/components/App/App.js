import React from "react";

import { Route, Switch } from "react-router-dom";
import Theme from "../../theme/Theme";
import { Wrapper } from "./AppStyle";
import { Circle1, Circle2 } from "../Shared/BackgroundCircle/BackgroundCircles";

import Header from "../Shared/Header";
import Home from "../Home/Home";
import Information from "../Information/Info";
import Page404 from "../PageNotFound/Page404";

// Routing tutorial: https://medium.com/javascript-in-plain-english/how-to-add-react-router-to-your-react-app-bf764a776122
function App() {
  return (
    <Theme>
      <Wrapper>
        <Circle1 />
        <Circle2 />
        <Header />
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/info" component={Information} />
          <Route path="*" component={Page404} />
        </Switch>
      </Wrapper>
    </Theme>
  );
}

export default App;
