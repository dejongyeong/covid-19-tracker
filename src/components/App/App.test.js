import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import { render, act } from "@testing-library/react";

import App from "./App";

// Tutorial 1: https://reactrouter.com/web/guides/testing
// Tutorial 2: https://testing-library.com/docs/example-react-router/
test("Home renders correctly", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Navigation", () => {
  const root = document.createElement("div");
  document.body.appendChild(root);

  render(
    <MemoryRouter initialEntries={["/"]}>
      <App />
    </MemoryRouter>,
    root
  );

  // initial page
  expect(document.getElementById("covid-data")).toBeInTheDocument();

  // verify page content after navigation
  act(() => {
    const goInfoLink = document.querySelector(".info-link");
    goInfoLink.dispatchEvent(new MouseEvent("click", { bubbles: true }));
  });

  expect(document.getElementById("covid-info")).toBeInTheDocument();
});

test("Landing on a bad page", () => {
  const root = document.createElement("div");
  document.body.appendChild(root);
  render(
    <MemoryRouter initialEntries={["/admin"]}>
      <App />
    </MemoryRouter>,
    root
  );

  expect(document.getElementById("page-404")).toBeInTheDocument();
});
