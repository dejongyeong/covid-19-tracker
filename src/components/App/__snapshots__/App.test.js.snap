// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="sc-bdvvtL bVewFv"
>
  <div
    className="sc-gsDKAQ jZUonc"
  />
  <div
    className="sc-dkPtRN iFhKdz"
  />
  <header
    className="MuiPaper-root MuiAppBar-root WithStyles(ForwardRef(AppBar))-root-3 MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        className="sc-hKwDye kJcUYp"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <img
            alt="Logo"
            src="logo.svg"
          />
        </a>
      </div>
      <div
        className="sc-eCImPb eTOskA"
      >
        <a
          aria-current="page"
          className="home-link active"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "#d62828",
            }
          }
        >
          Home
        </a>
        <a
          aria-current={null}
          className="info-link"
          href="/info"
          onClick={[Function]}
        >
          <button
            className="sc-jRQBWg dmdrSt"
          >
            Information
          </button>
        </a>
      </div>
      <div
        className="sc-gKclnd eZKPIm"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-4 MuiIconButton-edgeEnd"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </header>
  <div>
    <div
      className="sc-bkkeKt eQjasW"
      id="covid-data"
    >
      <div
        className="sc-hBUSln iLUidA"
        id="api-error"
      >
        <div
          className="sc-fotOHu cMAMha"
        >
          <svg
            className="spinner"
            fill="currentColor"
            height="1em"
            icon="spinner"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
            />
          </svg>
        </div>
        <div
          className="sc-fFeiMQ chGJUV"
        >
          <h1>
            Fetching Data...
          </h1>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-bYoBSM hIWaMD"
    id="footer"
  >
    <p>
      Design and Built by De Jong Yeong Â© 
      2021
       | Data Source: 
      <a
        href="https://disease.sh/"
      >
        Open Disease Data API
      </a>
    </p>
  </div>
</div>
`;
